typing/ctype.ml

```diff
@@ -2833,16 +2843,21 @@ and unify_row env row1 row2 =
     else None
   in
   let row0 = {row_fields = []; row_more = more; row_bound = ();
-              row_closed = closed; row_fixed = fixed; row_name = name} in
+              row_closed = closed; row_fixed=fixed; row_name = name} in
   let set_more row rest =
     let rest =
       if closed then
         filter_row_fields row.row_closed rest
       else rest in
-    if rest <> [] && (row.row_closed || row_fixed row)
-    || closed && row_fixed row && not row.row_closed then begin
+    begin if rest <> [] && row.row_closed then
```

Comment:
At first glance the disappearance of the `|| row_fixed row` (or equivalent) part seems like a mistake.
Is it really, or can you justify that it is actually correct?

Reply:
You are right this was a major mistake. In particular, erroneous code like:
```OCaml
let f: 'a. ([> `A] as 'a) -> [> `A | `B | `C] = fun x -> x`
```
fell through the previous test.
The new code factorize the `row_fixed row` predicate without altering the full test.